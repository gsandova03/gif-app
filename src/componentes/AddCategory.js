import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [input, setInput] = useState("");
  
  const handleChange = (e) => {
    setInput(e.target.value);
  }
  
  const handleSubmit = (e) => {
    e.preventDefault();
    // console.log("HandleSubmit se disparo")
  
    if( input.trim().length > 2 ){
      setCategories(( categories ) => [ input ,...categories ]);
      setInput('');
    }
  
  }

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ input }</p>
      <input 
        type='text'
        value={ input }
        onChange={ handleChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,

}