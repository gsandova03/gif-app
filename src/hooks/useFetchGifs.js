import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = ( category ) => {

  const [images, setImages] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGifs( category )
      .then( (imgs) => {
        setImages({
          data: imgs,
          loading: false
        })
      })
  }, [ category ])

  return images; 
}
