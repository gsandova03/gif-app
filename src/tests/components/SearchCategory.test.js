import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { SearchCategory } from '../../componentes/SearchCategory';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente de <SearchCategory />', ()=>{
  const category = "One Punch"
  
  
  
  test('debe de mostrar el componente correctamente ', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow( <SearchCategory category={ category }/> )
    expect( wrapper ).toMatchSnapshot();
  })
  
  test('debe de mostrar items cuando recibe informacion del useFetchGifs', () => {
    
    //mock para fingir que nuestro componente tiene informacion

    const imgs = [
      {
        id: 'abc',
        url: 'https://localhost.com',
        title: 'Hola mundo'
      },
      {
        id: 'def',
        url: 'https://localhost.com',
        title: 'Hola mundo'
      }
    ]

    //Agregar valores simulados
    useFetchGifs.mockReturnValue({
      data: imgs,
      loading: false
    });

    const wrapper = shallow( <SearchCategory category={ category }/> );
    
    expect( wrapper.find('p').exists() ).toBe(false)
    expect( wrapper.find('GifGridItem').length ).toBe( imgs.length )
  })
  
})