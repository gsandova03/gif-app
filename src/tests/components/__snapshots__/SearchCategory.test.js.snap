// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de <SearchCategory /> debe de mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <p
      className=" animate__animated animate__pulse"
    >
      Cargando ...
    </p>
  </div>
</Fragment>
`;
